src/cli/cmd/add/flake.rs

```diff
@@ -29,17 +31,28 @@ pub(crate) fn update_flake_input(
     flake_input_value: url::Url,
     flake_contents: String,
 ) -> color_eyre::Result<String> {
-    let nixel::Expression::String(existing_input_value) = *attr.to else {
-        // OK technically it can be an IndentedString but I don't want to support that so yeah
-        return Err(color_eyre::eyre::eyre!(
-            "`inputs.{flake_input_name}.url` was not a string" // this is enforced by Nix itself
-        ))?;
-    };
-    replace_input_value(
-        &existing_input_value.parts,
-        &flake_input_value,
-        &flake_contents,
-    )
+    match *attr.to {
+        nixel::Expression::String(existing_input_value) => replace_input_value_string(
+            &existing_input_value.parts,
+            &flake_input_value,
+            &flake_contents,
+        ),
+        nixel::Expression::IndentedString(existing_input_value) => replace_input_value_string(
+            &existing_input_value.parts,
+            &flake_input_value,
+            &flake_contents,
+        ),
+        nixel::Expression::Uri(existing_input_value) => {
+            replace_input_value_uri(&existing_input_value, &flake_input_value, &flake_contents)
+        }
+        otherwise => {
+            // Unclear what this case could be
```

Comment:
This is any other expression type in the Nix language (i.e., it could be a boolean, a number, or even another attrset.  They would all be disallowed by Nix itself, so if this branch is ever hit, it's likely not something we could (or should) try to fix.